### PR TITLE
Fix static scan issues

### PR DESCRIPTION
**Pull request by xianju6x**
Tracked-On: OAM-110031

### CODE REVIEW

File: i2c.c [hunk 1]
```diff
@@ -74,9 +74,10 @@ int write_i2c_device(char bus, int addr, int reg, int size, int value)
 	}
 
 	ret = ioctl(fd,  DEF_I2C_SLAVE_FORCE, addr);
-	if (fd == -1) {
+	if (ret == -1) {
 		printf("Error with the ioctl for address 0x%x\n", addr);
-		return -1;
+		ret = -1;
```
Comment (jingdlu) [inline i2c.c]: Don't need this ret = -1 since you have checked ret value.

File: i2c.c [hunk 2]
```diff
@@ -132,14 +134,16 @@ int block_write_i2c_device(char bus, int addr, int reg, int size, uint8_t array_
...
```
Comment (jingdlu) [inline i2c.c]: Don't need this line.

File: i2c.c [hunk 3]
```diff
@@ -185,7 +190,8 @@ int read_i2c_device(char bus, int addr, int reg, int size, int *result)
 	ret = ioctl(fd,  DEF_I2C_SLAVE_FORCE, addr);
 	if (ret == -1) {
 		printf("Error with the ioctl for address 0x%x\n", addr);
-		return -1;
+		ret = -1;
```
Comment (jingdlu) [inline i2c.c]: Don't need this line.

File: i2c.c [hunk 4]
```diff
@@ -248,7 +258,8 @@ int block_read_i2c_device(char bus, int addr, int reg, int size, uint8_t array_s
...
```
Comment (jingdlu) [inline i2c.c]: Don't need this line.

File: peeknpoke.c [hunk 5]
```diff
@@ -166,7 +166,7 @@ static int process_i2c_args(int argc, char **argv)
 		printf("\n***Invalid I2C operation\n");
 		status = -1;
 	}
-
+	free(values);
```
Comment (jingdlu) [inline peeknpoke.c]: Don't add free(values) here. Because malloc doesn't happen for every case. You should free values when values usage is done.

**Review by jingdlu — APPROVED** [review]

**Review by kwang13 — APPROVED** [review]

**Pull request merged**